acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -17,6 +17,8 @@ pub(crate) enum BrilligArithmeticError {
     MismatchedRhsBitSize { rhs_bit_size: u32, op_bit_size: u32 },
     #[error("Attempted to shift by {shift_size} bits on a type of bit size {bit_size}")]
     BitshiftOverflow { bit_size: u32, shift_size: u32 },
+    #[error("Attempted to shift by more than {bit_size} bits on a type of bit size {bit_size}")]
+    BitshiftOverflowBitSize { bit_size: u32 },
```

Comment:
these look redundant.

Reply:
I see that some tests are failing because they expected `BitshiftOverflow` and got `BitshiftOverflowBitSize`. 

Since it might be a niche error, maybe you could just use `BitshiftOverflow { shift_size: u32::MAX, ...`. That would indicate to people that they are dealing with some huge number they probably didn't intend, and it could be even more obvious than saying "attempted to shift by more than 32 bits", because they might think, hm, was it 64, or 33?

Reply:
I'd be in favour of just standardising on a single error variant for this. I haven't looked in detail on why this was added however.

Reply:
My understanding was that if the `shift_size` is over `u32::MAX` we get the new error which doesn't have the `shift_size` field, otherwise the old error which does.

Reply:
I finally fixed this by using a u128 in the original error, so it can display properly all the cases.